File: clojure/log-levels/src/log_levels.clj
(ns log-levels
  (:require [clojure.string :as str]))

(defn message [s]
  (str/trim (last (str/split s #":"))))

(defn log-level [s]
  (str/lower-case (second (re-find #"^\[(.+)\]" s))))  

(defn reformat [s]
  (str (message s) " (" (log-level s) ")"))

File: cpp/log-levels/log_levels.cpp
#include <string>

namespace log_line
{
    std::string message(std::string log_line)
    {
        return log_line.substr(log_line.find(' ') + 1);
    }

    std::string log_level(std::string log_line)
    {
        return log_line.substr(1, log_line.find(']') - 1);
    }

    std::string reformat(std::string log_line)
    {
        return message(log_line) + " (" + log_level(log_line) + ")";
    }
} // namespace log_line
